Fix progress and improvements.

jjnicola opened the pull request:
- Use enum for scan status.
- Update the test accordingly.
- Remove legacy_target string from the scan_info table.
- Fix progress calculation for legacy target mode.
- Check for scan status before trying to deleted.

jjnicola committed:
Use enum for scan status.
Add new status init for the new scan still not launched.
Update the test accordingly.

jjnicola committed:
Remove legacy_target string from the scan_info table.
This improve target handling and make easier to work with legacy targets.
Update tests accordingly.

jjnicola committed:
Fix progress calculation for legacy target mode.
Also fix an issue which finished target forced others scans to stop.

jjnicola committed:
Check for scan status before trying to deleted.

jjnicola committed:
Solve issues reported by Codacy.

bjoernricks commented on ospd/misc.py:
```diff
@@ -135,26 +143,26 @@ def ids_iterator(self):
         if self.data_manager is None:
             self.data_manager = multiprocessing.Manager()
+        if not options:
+            options = Dict()
```
```suggestion
            options = dict()
```

bjoernricks commented on ospd/misc.py:
```diff
@@ -185,10 +193,15 @@ def get_progress(self, scan_id):
-        return self.scans_table[scan_id]['target_progress']
+        total_hosts = len(target_str_to_list(target))
+        host_progresses = self.scans_table[scan_id]['target_progress'].get(target)
+        t_prog = sum(host_progresses.values()) / total_hosts
```
Could you check and ensure that total_hosts will never be zero?

bjoernricks commented on ospd/misc.py:
```diff
@@ -202,14 +215,11 @@ def get_end_time(self, scan_id):
 
     def get_target(self, scan_id):
```
Personally I would rename the method to `get_targets` or `get_target_list`.

bjoernricks reviewed (changes requested):
Only one small mistake Dict instead of dict

jjnicola committed:
Some fix.
Check for division by zero error.
Fix typo for dict().
Rename get_target() to get_target_list().
